guides/testing-with-test-dispatch.md

```diff
@@ -0,0 +1,76 @@
+# Testing with TestDispatch
+
+This library was build to test if forms and links have the expected results in
+controller tests.
+
+It can be used with or without
+[TestSelector](https://github.com/DefactoSoftware/TestSelector)
+
+## Testing a form
+
+We will explain the workings with the following form.
+
+```html
+<form action="admin/users/create" method="post">
+  <input name="_csrf_token" type="hidden" value="AHIqJmEUAxIvGy4HJ3oUGCMjChsLYBZ-SGgy7W1HElh3PKLsffgXXQO6">
+
+  <label>Name</label>
+  <input id="user_name" label="Name" name="user[name]" type="text">
+
+  <label>Email</label>
+  <input id="user_email" label="Email" name="user[email]" type="email">
+
+  <label>Description</label>
+  <textarea id="user_description" label="Description" name="user[description]" type="textarea"></textarea>
+
+  <select id="user_role" name="user[role]">
+    <option value="">Select a role</option>
+    <option value="admin">Admin</option>
+    <option value="moderator">Moderator</option>
+  </select>
+
+  <button type="submit">Create new user</button>
+</form>
+```
+
+This form is shown on `/admin/users/new` the entity for this form is `user`.
+
+To post the form as is we can use the following code.
+
+```elixir
+conn
+|> get("/admin/users/new")
+|> disptach_form_with(:user)
+|> html_response(422) =~ "can't be blank"
+```
+
+There is now a request made to the router with the method `POST` to the action
+`/admin/users/create` and with the params %{name: "", email: "", description:
+"", roles: ""}
```

Comment:
```suggestion
`/admin/users/create` and with the params `%{name: "", email: "", description:
"", roles: ""}`
```

Reply:
Nice, this needs some more changes since it’s scoped in the user map. And we might change this to the => syntax